Carbon.IIS/Carbon.IIS.psm1

```diff
@@ -14,12 +14,54 @@
 
 #Requires -Version 5.1
 Set-StrictMode -Version 'Latest'
+$InformationPreference = 'Continue'
 
 # Functions should use $moduleRoot as the relative root from which to find
 # things. A published module has its function appended to this file, while a 
 # module in development has its functions in the Functions directory.
 $moduleRoot = $PSScriptRoot
 
+if( [Environment]::SystemDirectory )
+{
+    $microsoftWebAdministrationPath =
+        Join-Path -Path ([Environment]::SystemDirectory) -ChildPath 'inetsrv\Microsoft.Web.Administration.dll'
+    if( $PSVersionTable['PSVersion'].Major -eq 6 )
+    {
+        $microsoftWebAdministrationPath = Join-Path -Path $moduleRoot -ChildPath 'bin\Microsoft.Web.Administration.dll'
+    }
+    
+    if( (Test-Path -Path $microsoftWebAdministrationPath) )
+    {
+        Add-Type -Path $microsoftWebAdministrationPath
+    }
+}
+
+if( -not (Test-CTypeDataMember -TypeName 'Microsoft.Web.Administration.Site' -MemberName 'PhysicalPath') )

```

Comment:
Where does `Test-CTypeDataMember` come from? I don't see any nested modules being imported.

Reply:
Whoops. That's from Carbon. I'll take care of it.

Reply:
PR for Carbon.Core so it contains this and other type data helper functions: https://github.com/webmd-health-services/Carbon.Core/pull/3

Reply:
Reviewed.

Reply:
Carbon.Core is now included as a private module.